### ISSUE TITLE
Bug: Exception is being thrown for the Type level PropertyHandler for Dynamic Insertion

### ISSUE THREAD

**Issue by mikependon** [mikependon/RepoDB#628]
An exception is being thrown if the `Insert` operation is called with anonymous types, like below.

```csharp
connection.Insert("Table", new { Id = 1, ColumnDateTime = SomeDateValue });
```

And if you defined a property handler on the type level of the `DateTime`.

```csharp
FluentMapper
    .Type<DateTime>()
    .PropertyHandler<DateTimeKindUtcPropertyHandler>();
```

Error Image:

![image](https://user-images.githubusercontent.com/591015/97310035-1484ab80-1863-11eb-8d44-e56bcdb14b08.png)

**Comment by mikependon** [comment]
Could not replicate this on our environment using the RepoDB v1.12.4 version. The code below is working.

Database Table:

```csharp
CREATE TABLE [dbo].[Person](
	[Id] [bigint] IDENTITY(1,1) NOT NULL,
	[Name] [nvarchar](128) NOT NULL,
	[Age] [int] NOT NULL,
	[Address] [nvarchar](max) NOT NULL,
	[IsActive] [bit] NOT NULL,
	[CreatedDateUtc] [datetime2](5) NOT NULL,
 CONSTRAINT [PK__Person__3214EC070E314C0E] PRIMARY KEY CLUSTERED 
(
	[Id] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY] TEXTIMAGE_ON [PRIMARY]
GO
```

PropertyHandler:

```csharp
using RepoDb;
using RepoDb.Interfaces;
using System;

namespace AnonymousTypePropertyHandler.PropertyHandlers
{
    public class DateTimeToUtcKindHandler : IPropertyHandler<DateTime?, DateTime?>
    {
        public DateTime? Get(DateTime? input,
            ClassProperty property)
        {
            return input.HasValue ?
                DateTime.SpecifyKind(input.Value, DateTimeKind.Utc) :
                (DateTime?)null;
        }

        public DateTime? Set(DateTime? input,
            ClassProperty property)
        {
            return input.HasValue ?
                DateTime.SpecifyKind(input.Value, DateTimeKind.Unspecified) :
                (DateTime?)null;
        }
    }
}
```

CSharp Application:

```csharp
using AnonymousTypePropertyHandler.PropertyHandlers;
using Microsoft.Data.SqlClient;
using RepoDb;
using System;

namespace AnonymousTypePropertyHandler
{
    class Program
    {
        static void Main(string[] args)
        {
            Bootstrap();
            Test();
        }

        static void Bootstrap()
        {
            SqlServerBootstrap.Initialize();
            FluentMapper
                .Type<DateTime>()
                .PropertyHandler<DateTimeToUtcKindHandler>(true);
        }

        static void Test()
        {
            using (var connection = new SqlConnection("Server=.;Database=TestDB;Integrated Security=SSPI;"))
            {
                var id = connection.Insert("Person", new
                {
                    Name = "John Doe",
                    Age = 30,
                    Address = "New York",
                    IsActive = true,
                    CreatedDateUtc = DateTime.UtcNow
                });
            }
        }
    }
}
```

**Comment by mikependon** [comment]
I have replicated the issue in this context. The error is being raised if the TResult is defined during the execution of the Insert operation.

**No issue:**

```csharp
var id = connection.Insert("Person", new
{
    Name = "John Doe",
    Age = 30,
    Address = "New York",
    IsActive = true,
    CreatedDateUtc = DateTime.UtcNow
});
```

**With issue:**

```csharp
var id = connection.Insert<int>("Person", new
{
    Name = "John Doe",
    Age = 30,
    Address = "New York",
    IsActive = true,
    CreatedDateUtc = DateTime.UtcNow
});
```

**Comment by mikependon** [comment]
The fix for this is available at [RepoDb v1.12.5-beta2](https://www.nuget.org/packages/RepoDb/1.12.5-beta2).
